File: src/test/java/io/binac/leetcode/WordBreakTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class WordBreakTest {
    private final WordBreak.Solution1 solution1 = new WordBreak.Solution1();

    private final String s1 = "leetcode";
    private final List<String> wordDict1 = Arrays.asList("leet", "code");

    private final String s2 = "applepenapple";
    private final List<String> wordDict2 = Arrays.asList("apple", "pen");

    private final String s3 = "catsandog";
    private final List<String> wordDict3 = Arrays.asList("cats", "dog", "sand", "and", "cat");

    @Test
    void test1() {
        assertTrue(solution1.wordBreak(s1, wordDict1));
        assertTrue(solution1.wordBreak(s2, wordDict2));
        assertFalse(solution1.wordBreak(s3, wordDict3));
    }
}

File: cpp/unittests/WordBreakTest.cpp
#include "WordBreak.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(WordBreak, Solution139_1) {
  const std::string S1 = "leetcode";
  std::vector<std::string> WordDict1{"leet", "code"};
  EXPECT_TRUE(Solution132_1().wordBreak(S1, WordDict1));

  const std::string S2 = "applepenapple";
  std::vector<std::string> WordDict2{"apple", "pen"};
  EXPECT_TRUE(Solution132_1().wordBreak(S2, WordDict2));

  const std::string S3 = "catsandog";
  std::vector<std::string> WordDict3{"cats", "dog", "sand", "and", "cat"};
  EXPECT_FALSE(Solution132_1().wordBreak(S3, WordDict3));
}
